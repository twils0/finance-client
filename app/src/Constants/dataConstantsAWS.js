// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
