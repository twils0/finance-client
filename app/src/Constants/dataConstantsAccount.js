export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
