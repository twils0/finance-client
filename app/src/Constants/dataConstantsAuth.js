export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
