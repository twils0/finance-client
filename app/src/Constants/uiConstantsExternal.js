export const actionTypes = {
  SET_CURRENT_FORM: 'EXTERNAL_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'EXTERNAL_SET_FORM_HEIGHT',
  SET_INPUT_VALUE_ERROR: 'EXTERNAL_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'EXTERNAL_SET_BUTTON_TEXT',
};

export const formNames = {
  RESET_PASSWORD: 'resetPasswordForm',
  FORGOT_PASSWORD: 'forgotPasswordForm',
  LOGIN: 'loginForm',
  CARD: 'cardForm',
  INFO: 'infoForm',
  SIGN_UP: 'signUpForm',
  CODE_MFA_PHONE: 'codePhoneMFAForm',
  CODE_VERIFY_PHONE: 'codePhoneVerifyForm',
  CODE_VERIFY_EMAIL: 'codeEmailVerifyForm',
  DEVICE: 'deviceForm',
};

export const inputNames = {
  [formNames.RESET_PASSWORD]: {
    CODE: 'codeResetPassword',
    PASSWORD: 'passwordResetPassword',
    PASSWORD2: 'password2ResetPassword',
  },
  [formNames.FORGOT_PASSWORD]: {
    EMAIL: 'loginPasswordEmail',
  },
  [formNames.LOGIN]: {
    EMAIL: 'loginEmail',
    PASSWORD: 'loginPassword',
  },
  [formNames.CARD]: {
    NAME_ON_CARD: 'cardNameOnCard',
    PROMO_CODE: 'cardPromoCode',
    STRIPE: 'cardStripe',
  },
  [formNames.INFO]: {
    NAME: 'infoName',
    PHONE: 'infoPhone',
    EMAIL: 'infoEmail',
  },
  [formNames.SIGN_UP]: {
    PASSWORD: 'signUpPassword',
    PASSWORD2: 'signUpPassword2',
  },
  [formNames.CODE_MFA_PHONE]: {
    CODE: 'codeMFAPhoneCode',
  },
  [formNames.CODE_VERIFY_PHONE]: {
    CODE: 'codeVerifyPhoneCode',
  },
  [formNames.CODE_VERIFY_EMAIL]: {
    CODE: 'codeVerifyEmailCode',
  },
};

export const buttonNames = {
  FIRST: 'firstButton',
  SECOND: 'secondButton',
  FORGOT_PASSWORD: 'forgotPasswordButton',
  RESEND_PHONE: 'resendPhoneButton',
  RESEND_EMAIL: 'resendEmailButton',
};

export const buttonTexts = {
  RESET: 'Reset',
  CANCEL: 'Cancel',
  SEND_CODE: 'Send Code',
  RESEND_PHONE: 'Resend SMS',
  RESEND_EMAIL: 'Resend Email',
  LOG_IN: 'Log In',
  FORGOT_PASSWORD: 'Forgot Password',
  SIGN_UP: 'Sign Up',
  NEXT: 'Next',
  PREVIOUS: 'Previous',
  SUBMIT: 'Submit',
  YES: 'Yes',
  NO: 'No',
  CONTINUE: 'Continue',
};
