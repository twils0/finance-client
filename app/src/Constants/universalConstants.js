import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
