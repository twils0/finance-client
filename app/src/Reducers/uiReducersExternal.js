import { combineReducers } from 'redux';

import {
  formNames,
  inputNames,
  buttonNames,
  buttonTexts,
  actionTypes,
} from '../Constants/uiConstantsExternal';

export const initialState = {
  utilities: {
    forms: {
      springSettings: { stiffness: 170, damping: 26 },
      width: 300,
      widthBuffer: 60,
      mediaWidth: 700,
      heightBuffer: 50,
      inputs: {
        heightBuffer: 15,
      },
    },
    buttons: {
      width: 300,
      heightBuffer: 20,
      heightBottomBuffer: 30,
    },
    terms: {
      bufferPage: 30,
      bufferHeightParagraphTop: 20,
      bufferHeightParagraphBottom: 30,
      bufferHeight: 20,
    },
  },
  forms: {
    current: formNames.LOGIN,
    list: [...Object.values(formNames)],
    [formNames.RESET_PASSWORD]: {
      id: formNames.RESET_PASSWORD,
      height: 339,
      inputsList: [...Object.values(inputNames[formNames.RESET_PASSWORD])],
      inputs: {
        [inputNames[formNames.RESET_PASSWORD].CODE]: {
          parentId: formNames.RESET_PASSWORD,
          id: inputNames[formNames.RESET_PASSWORD].CODE,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.RESET_PASSWORD].PASSWORD]: {
          parentId: formNames.RESET_PASSWORD,
          id: inputNames[formNames.RESET_PASSWORD].PASSWORD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.RESET_PASSWORD].PASSWORD2]: {
          parentId: formNames.RESET_PASSWORD,
          id: inputNames[formNames.RESET_PASSWORD].PASSWORD2,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.FORGOT_PASSWORD]: {
      id: formNames.FORGOT_PASSWORD,
      height: 220,
      inputsList: [...Object.values(inputNames[formNames.FORGOT_PASSWORD])],
      inputs: {
        [inputNames[formNames.FORGOT_PASSWORD].EMAIL]: {
          parentId: formNames.FORGOT_PASSWORD,
          id: inputNames[formNames.FORGOT_PASSWORD].EMAIL,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.LOGIN]: {
      id: formNames.LOGIN,
      height: 222,
      inputsList: [...Object.values(inputNames[formNames.LOGIN])],
      inputs: {
        [inputNames[formNames.LOGIN].EMAIL]: {
          parentId: formNames.LOGIN,
          id: inputNames[formNames.LOGIN].EMAIL,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.LOGIN].PASSWORD]: {
          parentId: formNames.LOGIN,
          id: inputNames[formNames.LOGIN].PASSWORD,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CARD]: {
      id: formNames.CARD,
      height: 360,
      inputsList: [...Object.values(inputNames[formNames.CARD])],
      inputs: {
        [inputNames[formNames.CARD].NAME_ON_CARD]: {
          parentId: formNames.CARD,
          id: inputNames[formNames.CARD].NAME_ON_CARD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.CARD].PROMO_CODE]: {
          parentId: formNames.CARD,
          id: inputNames[formNames.CARD].PROMO_CODE,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.CARD].STRIPE]: {
          parentId: formNames.CARD,
          id: inputNames[formNames.CARD].STRIPE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.INFO]: {
      id: formNames.INFO,
      height: 280,
      inputsList: [...Object.values(inputNames[formNames.INFO])],
      inputs: {
        [inputNames[formNames.INFO].NAME]: {
          parentId: formNames.INFO,
          id: inputNames[formNames.INFO].NAME,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.INFO].EMAIL]: {
          parentId: formNames.INFO,
          id: inputNames[formNames.INFO].EMAIL,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.INFO].PHONE]: {
          parentId: formNames.INFO,
          id: inputNames[formNames.INFO].PHONE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.SIGN_UP]: {
      id: formNames.SIGN_UP,
      height: 247,
      inputsList: [...Object.values(inputNames[formNames.SIGN_UP])],
      inputs: {
        [inputNames[formNames.SIGN_UP].PASSWORD]: {
          parentId: formNames.SIGN_UP,
          id: inputNames[formNames.SIGN_UP].PASSWORD,
          value: '',
          errorMessage: '',
        },
        [inputNames[formNames.SIGN_UP].PASSWORD2]: {
          parentId: formNames.SIGN_UP,
          id: inputNames[formNames.SIGN_UP].PASSWORD2,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CODE_MFA_PHONE]: {
      id: formNames.CODE_MFA_PHONE,
      height: 183,
      inputsList: [...Object.values(inputNames[formNames.CODE_MFA_PHONE])],
      inputs: {
        [inputNames[formNames.CODE_MFA_PHONE].CODE]: {
          parentId: formNames.CODE_MFA_PHONE,
          id: inputNames[formNames.CODE_MFA_PHONE].CODE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CODE_VERIFY_PHONE]: {
      id: formNames.CODE_VERIFY_PHONE,
      height: 183,
      inputsList: [...Object.values(inputNames[formNames.CODE_VERIFY_PHONE])],
      inputs: {
        [inputNames[formNames.CODE_VERIFY_PHONE].CODE]: {
          parentId: formNames.CODE_VERIFY_PHONE,
          id: inputNames[formNames.CODE_VERIFY_PHONE].CODE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.CODE_VERIFY_EMAIL]: {
      id: formNames.CODE_VERIFY_EMAIL,
      height: 183,
      inputsList: [...Object.values(inputNames[formNames.CODE_VERIFY_EMAIL])],
      inputs: {
        [inputNames[formNames.CODE_VERIFY_EMAIL].CODE]: {
          parentId: formNames.CODE_VERIFY_EMAIL,
          id: inputNames[formNames.CODE_VERIFY_EMAIL].CODE,
          value: '',
          errorMessage: '',
        },
      },
    },
    [formNames.DEVICE]: {
      id: formNames.DEVICE,
      height: 146,
      inputsList: [],
      inputs: {},
    },
  },
  buttons: {
    list: [...Object.values(buttonNames)],
    [buttonNames.FIRST]: { id: buttonNames.FIRST, text: buttonTexts.LOG_IN },
    [buttonNames.SECOND]: { id: buttonNames.SECOND, text: buttonTexts.SIGN_UP },
    [buttonNames.FORGOT_PASSWORD]: {
      id: buttonNames.FORGOT_PASSWORD,
      text: buttonTexts.FORGOT_PASSWORD,
    },
    [buttonNames.RESEND_PHONE]: { id: buttonNames.RESEND_PHONE, text: buttonTexts.RESEND_PHONE },
    [buttonNames.RESEND_EMAIL]: { id: buttonNames.RESEND_EMAIL, text: buttonTexts.RESEND_EMAIL },
  },
};

export const utilities = (state = initialState.utilities) => state;

export const forms = (state = initialState.forms, action) => {
  switch (action.type) {
    case actionTypes.SET_CURRENT_FORM:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.SET_FORM_HEIGHT:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_INPUT_VALUE_ERROR:
      return {
        ...state,
        [action.payload.parentId]: {
          ...state[action.payload.parentId],
          inputs: {
            ...state[action.payload.parentId].inputs,
            [action.payload.id]: {
              ...state[action.payload.parentId].inputs[action.payload.id],
              ...action.payload,
            },
          },
        },
      };
    default:
      return state;
  }
};

export const buttons = (state = initialState.buttons, action) => {
  switch (action.type) {
    case actionTypes.SET_BUTTON_TEXT:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    default:
      return state;
  }
};

const uiReducersExternal = combineReducers({
  utilities,
  forms,
  buttons,
});

export default uiReducersExternal;
