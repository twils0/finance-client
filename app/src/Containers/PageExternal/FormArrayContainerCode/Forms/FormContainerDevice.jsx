import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { BeatLoader } from 'halogenium';

import theme from '../../../../themes';

import { requestStatusTypes } from '../../../../Constants/universalConstants';
import { statusNames } from '../../../../Constants/dataConstantsAuth';
import { formNames } from '../../../../Constants/uiConstantsExternal';

import FlexColumn from '../../../../Components/Flex/FlexColumn';
import FormDevice from '../../../../Components/PageExternal/FormArrayCode/Forms/FormDevice';

class FormContainerDevice extends React.Component {
  componentDidMount() {
    this.props.setHeight(formNames.DEVICE);
  }

  componentWillUpdate(nextProps) {
    const { statusAuth } = nextProps;

    if (
      statusAuth[statusNames.SIGN_OUT_DEVICES].status === requestStatusTypes.LOADING ||
      statusAuth[statusNames.SIGN_OUT_DEVICES].status === requestStatusTypes.SUCCESS
    ) {
      nextProps.clearElement(formNames.DEVICE);
    }
  }

  componentDidUpdate() {
    this.props.setHeight(formNames.DEVICE);
  }

  componentWillUnmount() {
    this.props.clearElement(formNames.DEVICE);
  }

  render() {
    const {
      heightRef,
      width,
      mediaWidth,
      formHeightBuffer,
      inputHeightBuffer,
      statusAuth,
      currentForm,
      inputs,
    } = this.props;

    if (
      statusAuth[statusNames.SIGN_OUT_DEVICES].status === requestStatusTypes.LOADING ||
      statusAuth[statusNames.SIGN_OUT_DEVICES].status === requestStatusTypes.SUCCESS
    ) {
      const widthString = `${width}px`;

      return (
        <FlexColumn width={widthString} height="100%">
          <BeatLoader color={theme.color.secondary.default} />
        </FlexColumn>
      );
    }
    return (
      <FormDevice
        heightRef={heightRef}
        width={width}
        mediaWidth={mediaWidth}
        formHeightBuffer={formHeightBuffer}
        inputHeightBuffer={inputHeightBuffer}
        currentFormBool={currentForm === formNames.DEVICE}
        inputs={inputs}
        handleInputChange={this.handleInputChange}
      />
    );
  }
}

FormContainerDevice.propTypes = {
  heightRef: PropTypes.func.isRequired,
  width: PropTypes.number.isRequired,
  mediaWidth: PropTypes.number.isRequired,
  formHeightBuffer: PropTypes.number.isRequired,
  inputHeightBuffer: PropTypes.number.isRequired,
  statusAuth: PropTypes.object.isRequired,
  currentForm: PropTypes.string.isRequired,
  inputs: PropTypes.object.isRequired,
  setHeight: PropTypes.func.isRequired,
  clearElement: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => ({
  heightRef: ownProps.heightRef,
  width: state.ui.external.utilities.forms.width,
  mediaWidth: state.ui.external.utilities.forms.mediaWidth,
  formHeightBuffer: state.ui.external.utilities.forms.heightBuffer,
  inputHeightBuffer: state.ui.external.utilities.forms.inputs.heightBuffer,
  statusAuth: state.data.auth.status,
  currentForm: state.ui.external.forms.current,
  inputs: state.ui.external.forms[formNames.DEVICE].inputs,
  setHeight: ownProps.setHeight,
  clearElement: ownProps.clearElement,
});

export default connect(mapStateToProps)(FormContainerDevice);
