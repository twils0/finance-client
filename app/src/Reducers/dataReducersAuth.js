import { combineReducers } from 'redux';

import { requestStatusTypes, pathNames } from '../Constants/universalConstants';
import { actionTypes, statusNames, codeTypeNames } from '../Constants/dataConstantsAuth';

export const initialState = {
  authenticated: false,
  redirectURL: pathNames.WATCHLIST,
  status: {
    [statusNames.LOGIN]: {
      id: statusNames.LOGIN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_OUT_DEVICES]: {
      id: statusNames.SIGN_OUT_DEVICES,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGIN_MFA]: {
      id: statusNames.LOGIN_MFA,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.SIGN_UP]: {
      id: statusNames.SIGN_UP,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE]: {
      id: statusNames.VERIFY_PHONE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_PHONE_CODE]: {
      id: statusNames.VERIFY_PHONE_CODE,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL]: {
      id: statusNames.VERIFY_EMAIL,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.VERIFY_EMAIL_LINK]: {
      id: statusNames.VERIFY_EMAIL_LINK,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.FORGOT_PASSWORD]: {
      id: statusNames.FORGOT_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.RESET_PASSWORD]: {
      id: statusNames.RESET_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.CHANGE_PASSWORD]: {
      id: statusNames.CHANGE_PASSWORD,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.LOGOUT]: {
      id: statusNames.LOGOUT,
      status: requestStatusTypes.SUCCESS,
    },
    [statusNames.DELETE_ACCOUNT]: {
      id: statusNames.DELETE_ACCOUNT,
      status: requestStatusTypes.SUCCESS,
    },
  },
  codeTypes: {
    [codeTypeNames.VERIFY_PHONE]: {
      id: codeTypeNames.VERIFY_PHONE,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL]: {
      id: codeTypeNames.VERIFY_EMAIL,
      needed: true,
    },
    [codeTypeNames.VERIFY_EMAIL_ADDITIONAL]: {
      id: codeTypeNames.VERIFY_EMAIL_ADDITIONAL,
      needed: true,
    },
  },
};

export const authenticated = (state = initialState.authenticated, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTHENTICATED:
      return action.payload.authenticated;
    case actionTypes.RESET_AUTH_STATE:
      return initialState.authenticated;
    default:
      return state;
  }
};

export const redirectURL = (state = initialState.redirectURL, action) => {
  switch (action.type) {
    case actionTypes.SET_REDIRECT_URL:
      return action.payload.redirectURL;
    default:
      return state;
  }
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AUTH_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const codeTypes = (state = initialState.codeTypes, action) => {
  switch (action.type) {
    case actionTypes.SET_CODE_TYPE:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_AUTH_STATE:
      return initialState.codeTypes;
    default:
      return state;
  }
};

const dataReducersAuth = combineReducers({
  authenticated,
  redirectURL,
  status,
  codeTypes,
});

export default dataReducersAuth;
