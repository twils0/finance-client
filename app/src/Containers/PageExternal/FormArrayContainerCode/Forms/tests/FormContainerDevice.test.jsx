import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import { initialState as initialStateAuth } from '../../../../../Reducers/dataReducersAuth';
import { initialState as initialStateUIExternal } from '../../../../../Reducers/uiReducersExternal';
import { requestStatusTypes } from '../../../../../Constants/universalConstants';
import { statusNames } from '../../../../../Constants/dataConstantsAuth';
import { formNames } from '../../../../../Constants/uiConstantsExternal';

import FormContainerDevice from '../FormContainerDevice';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const heightRef = jest.fn();
const setHeight = jest.fn();
const clearElement = jest.fn();

const shallowComponent = store =>
  shallow(
    <FormContainerDevice heightRef={heightRef} setHeight={setHeight} clearElement={clearElement} />,
    {
      context: {
        store,
      },
    },
  ).dive();

describe('Containers', () => {
  describe('PageExternal', () => {
    describe('FormArrayContainerCode', () => {
      describe('Forms', () => {
        describe('FormContainerDevice', () => {
          afterEach(() => {
            heightRef.mockReset();
            setHeight.mockReset();
            clearElement.mockReset();
          });

          it('shallow renders correctly', async () => {
            const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                external: stateBeforeUIExternal,
              },
            });

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
            expect(setHeight).toBeCalledWith(formNames.DEVICE);
          });

          it('shallow renders correctly, with componentWillUpdate, componentDidUpdate, and componentWillUnmount', async () => {
            const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));

            const store = mockStore({
              data: {
                auth: initialStateAuth,
              },
              ui: {
                external: stateBeforeUIExternal,
              },
            });

            const nextProps = {
              statusAuth: {
                [statusNames.SIGN_OUT_DEVICES]: {
                  status: requestStatusTypes.LOADING,
                },
              },
              clearElement,
            };

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const instance = wrapper.instance();

            instance.componentWillUpdate(nextProps);
            instance.componentDidUpdate();
            instance.componentWillUnmount();

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
            expect(setHeight).toBeCalledWith(formNames.DEVICE);
            expect(clearElement).toBeCalledWith(formNames.DEVICE);
            expect(setHeight).toBeCalledWith(formNames.DEVICE);
            expect(clearElement).toBeCalledWith(formNames.DEVICE);
            expect(setHeight).toHaveBeenCalledTimes(2);
            expect(clearElement).toHaveBeenCalledTimes(2);
          });

          it('shallow renders correctly, with loading indicator', async () => {
            const stateBeforeAuth = JSON.parse(JSON.stringify(initialStateAuth));
            const stateBeforeUIExternal = JSON.parse(JSON.stringify(initialStateUIExternal));

            stateBeforeAuth.status[statusNames.SIGN_OUT_DEVICES].status =
              requestStatusTypes.LOADING;

            const store = mockStore({
              data: {
                auth: stateBeforeAuth,
              },
              ui: {
                external: stateBeforeUIExternal,
              },
            });

            const expectedActions = [];

            const wrapper = shallowComponent(store);

            const actions = store.getActions();

            expect(actions).toEqual(expectedActions);
            expect(wrapper).toMatchSnapshot();
          });
        });
      });
    });
  });
});
