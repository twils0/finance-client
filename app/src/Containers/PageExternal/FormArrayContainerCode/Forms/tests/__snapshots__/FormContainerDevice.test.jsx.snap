// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers PageExternal FormArrayContainerCode Forms FormContainerDevice shallow renders correctly 1`] = `
<FormDevice
  currentFormBool={false}
  formHeightBuffer={50}
  heightRef={[MockFunction]}
  inputHeightBuffer={15}
  inputs={Object {}}
  mediaWidth={700}
  width={300}
/>
`;

exports[`Containers PageExternal FormArrayContainerCode Forms FormContainerDevice shallow renders correctly, with componentWillUpdate, componentDidUpdate, and componentWillUnmount 1`] = `
<FormDevice
  currentFormBool={false}
  formHeightBuffer={50}
  heightRef={[MockFunction]}
  inputHeightBuffer={15}
  inputs={Object {}}
  mediaWidth={700}
  width={300}
/>
`;

exports[`Containers PageExternal FormArrayContainerCode Forms FormContainerDevice shallow renders correctly, with loading indicator 1`] = `
<FlexColumn
  alignItems={null}
  alignSelf={null}
  borderColorType="border"
  borderRadius={null}
  borderStyle={null}
  borderWidth={null}
  boxSizing={null}
  color={Object {}}
  display={null}
  flex={null}
  flexWrap={null}
  height="100%"
  justifyContent={null}
  margin={null}
  mediaWidth={null}
  minHeight={null}
  minWidth={null}
  overflowX={null}
  overflowY={null}
  padding={null}
  position={null}
  responsiveAlignSecurities={null}
  responsiveJustifyContent={null}
  responsiveMargin={null}
  responsivePadding={null}
  responsivePosition={null}
  responsiveWidth={null}
  separator={false}
  themeColor="primary"
  width="300px"
>
  <BeatLoader
    color="#7367e6"
    loading={true}
    margin="2px"
    size="15px"
  />
</FlexColumn>
`;
